python/sglang/srt/mem_cache/hiradix_cache.py

```diff
@@ -361,12 +365,18 @@ def check_revoked_prefetch(self):
                 group=self.tp_group,
             )
         for _ in range(queue_size.item()):
-            req_id = self.cache_controller.prefetch_revoke_queue.get()
+            operation = self.cache_controller.prefetch_revoke_queue.get()
+            req_id = operation.request_id
             if req_id in self.ongoing_prefetch:
-                last_host_node, _, host_indices, _ = self.ongoing_prefetch[req_id]
+                last_host_node, _, host_indices, _, _ = self.ongoing_prefetch[req_id]
                 last_host_node.release_host()
                 self.cache_controller.mem_pool_host.free(host_indices)
                 del self.ongoing_prefetch[req_id]
+            else:
+                # Handle cases where check_prefetch_progress is executed prematurely
+                self.cache_controller.mem_pool_host.free(
```

Comment:
in such cases, memory should have been released already in `check_prefetch_progress`, this is double-free